### ISSUE TITLE
Parse.Query.toJSON() not working for aggregate query

### ISSUE THREAD

**Issue by mtrezza** [parse-community/Parse-SDK-JS#918]
### Issue Description

Parse.Query.toJSON() returns an empty query when the query is an aggregate query.

### Steps to reproduce

```
const query = new Parse.Query("MyCollection");
const pipeline = {
    "match":{"language":{"$in":[null,"en"]}},
};
const results = await query.aggregate(pipeline);
const queryJson = query.toJSON(); // returns `{where: {}}`
```
### Expected Results

`query.toJSON()` should return the aggregate query.

### Actual Outcome

`query.toJSON()` returns `{where: {}}`.

### Environment Setup

- **Server**
  - parse-server version (Be specific! Don't say 'latest'.) : 3.8.0
  - Operating System:     -
  - Hardware:             -
  - Localhost or remote server? (AWS, Heroku, Azure, Digital Ocean, etc): -

- **JS SDK**
  - JS SDK version: 2.7.0
  - Application? (Browser, Node, React-Native, etc): -

### Logs/Trace
-

**Comment by dplewis** [comment]
You want to implement `query.setPipeline()`?

**Comment by mtrezza** [comment]
Actually I will close this for now if that's OK, there may more important features to work on. At least we have it documented here.

**Comment by RaschidJFR** [comment]
@mtrezza what's your use case for this feature?

**Comment by mtrezza** [comment]
@RaschidJFR why?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by dplewis** [comment]
I forgot about this feature, I can do a PR if you want

**Comment by mtrezza** [comment]
No-one has voiced any need for it to my knowledge, and I personally don’t need it anymore, but it would certainly be a nice-to-have feature.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by RaschidJFR** [comment]
> @RaschidJFR why are you asking?
> 
> Sent with [GitHawk](http://githawk.com)

Because I've just started working on #1170 and I thought it might be related.

**Comment by mtrezza** [comment]
@RaschidJFR Related in what sense?

**Comment by mtrezza** [comment]
@RaschidJFR Related in what sense?

This issue is about the pipeline not being serialized with toJSON().
Your PR seems to be about mixing the where condition into the pipeline.

Can you please open an issue for your PR where you describe your intention, so the community can debate it if necessary? I already see some potential issues with that PR.

**Comment by RaschidJFR** [comment]
> Related in what sense?

@mtrezza I was just gathering information about similar problems to mine to try to solve them at once. Your issue is caused because `aggregate()` is executes the query but does not modify it, same as `find()` (they are sibling functions). So if you don't have a use for it anymore I'll just ignore it in my proposal.

I've created #1171 so let's move the conversation there. Please list the potential issues you've found.

**Comment by RaschidJFR** [comment]
> I forgot about this feature, I can do a PR if you want
@dplewis Please have a look at my PR #1170 and let me know if you need me to include anything else in case it goes ahead.

**Comment by zanderisrael** [comment]
I'd love to have this feature. I there something I can do to help this move along?

**Comment by mtrezza** [comment]
Thanks for asking @zanderisrael, you could open a PR. To my knowledge no-one has worked on it so far. It seems fairly straight forward, because a lot of the logic to produce the JSON output for non-aggregate queries already exists.

**Comment by zanderisrael** [comment]
Ok then, i'll have a look.

**Comment by zanderisrael** [comment]
Looks like I don't need this in the end. Someone else can pick this up...

**Comment by dplewis** [comment]
For this to work we would have to add `query.pipeline`

```
const query = new Parse.Query("MyCollection");
const pipeline = {
    "match":{"language":{"$in":[null,"en"]}},
};
const results = await query.aggregate(pipeline);
```
Becomes
```
const query = new Parse.Query("MyCollection");
query.pipeline({
    "match":{"language":{"$in":[null,"en"]}},
});
const results = await query.aggregate();
```
